backend/database/db.go

```diff
@@ -26,30 +30,112 @@ func Init() *gorm.DB {
 
 	DB, err = gorm.Open(dialector, &gorm.Config{})
 
-	// Migration/Creation of data tables for DB
-	// NOTE: Order of table creation matters
-	// Dependent tables must be created in the right
-	// order so that relations and keys exist.
-	if !DB.Migrator().HasTable(&models.User{}) {
-		DB.Migrator().CreateTable(&models.User{})
-	}
-	if !DB.Migrator().HasTable(&models.Session{}) {
-		DB.Migrator().CreateTable(&models.Session{})
-	}
-	if !DB.Migrator().HasTable(&models.Journey{}) {
-		DB.Migrator().CreateTable(&models.Journey{})
-	}
-	if !DB.Migrator().HasTable(&models.Event{}) {
-		DB.Migrator().CreateTable(&models.Event{})
+	// Grab command line arguments
+	// --migrate-db 			- Perform the DB migration
+	// --seed-db=some-file.json	- Seed the DB with a JSON file
+	doMigration := flag.Bool("migrate-db", false, "Do DB migration")
+	var seedFile string
+	flag.StringVar(&seedFile, "seed-db", "", "JSON File to seed the DB")
+	flag.Parse()
+
+	// Seed the DB
+	if seedFile != "" {
+		fmt.Println("--------------")
+		fmt.Println("Seeding the DB")
+		fmt.Println("--------------")
+		fmt.Println("File:", seedFile)
+
+		// Read Seed File
+		rawData, readErr := os.ReadFile(seedFile)
+		if readErr != nil {
+			panic(fmt.Sprintf("Seed file read failed: %s", readErr.Error()))
+		}
+		// Unmarshal Raw JSON file bytes
+		var seedData interface{}
+		unmarshalErr := json.Unmarshal(rawData, &seedData)
+		if unmarshalErr != nil { 
+		    panic(fmt.Sprintf("Seed JSON Unmarshal failed: %s", unmarshalErr.Error()))
+		}
+		seedJson := seedData.(map[string]interface{})
+
+		// Seed Users
+		users := seedJson["users"].([]interface{})
+		lenUsers := len(users)
+		fmt.Println("Number of users found:", lenUsers)
+
+		usersCreated := 0
+
+		for i := 0; i < lenUsers; i++ {
+			user := users[i].(map[string]interface{})
+
+			userUuid, uuidErr := uuid.Parse(user["id"].(string))
+			if uuidErr != nil {
+				panic(fmt.Sprintf("User ID parse failed: %s", uuidErr.Error()))
+			}
+
+			delete(user, "id") // Delete the ID, Store the rest in the "data" JSON blob
+			userMarshal, userMarshalErr := json.Marshal(user)
+			if userMarshalErr != nil {
+				panic(fmt.Sprintf("User Marshal failed: %s", userMarshalErr.Error()))
+			}
+			
+			newUser := models.User{UserID:userUuid, Data:userMarshal}
+			newUuid := uuid.New()
+			newUser.ID = newUuid
```

Comment:
We can likely combine this statement and the above I think into `newUser.ID = uuid.New()`